Update build instructions in README.md

gioneill opened the pull request:
Clarify some sticking points on the cURL and openSSL config steps.

winniequinn commented:
@gioneill I reworked the instructions for cURL and OpenSSL before merging. If you have any objections, please bonk me or make a new PR!

gioneill commented:
No objections!

On Tue, Dec 12, 2017 at 6:45 PM Winnie Quinn <notifications@github.com>
wrote:

> *@winniequinn* approved this pull request.
>
> I reworked the instructions for the OpenSSL and cURL sections and added
> info about the cURL version requirement. It looks good to me!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NYPL-Simplified/Simplified-iOS/pull/853#pullrequestreview-83021649>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF5Lb3EIEWn95eQuOUNToZxeM1Xj-ttOks5s_w_-gaJpZM4Q-I4O>
> .
>

winniequinn commented:
@gioneill If you can, please rename the branch before merging or make sure to adjust the merge title appropriately when merging. It'll be hard to remember what `gioneill-patch-1` was later when looking at commits in `master`.

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
 03. `git submodule update --init --recursive`
 04. Install [Cocoapods](https://cocoapods.org) if you haven't already.
 05. `pod install`
 06. Symlink the "DRM_Connector_Prerelease" directory to "adobe-rmsdk" within the "Simplified-iOS" directory. (You will need to have obtained the Adobe DRM Connector prerelease from Adobe.)
```
Do we want to be helpful with a sample command?  "ln -s [your_path_here/]DRM_Connector_Prerelease adobe-rmsdk"

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
 02. `cd Simplified-iOS`
 03. `git submodule update --init --recursive`
 04. Install [Cocoapods](https://cocoapods.org) if you haven't already.
 05. `pod install`
```
I also installed the Command Line Tools, while I was trying to figure out how to build libcurl.  I think I remember you telling me that I needed to install something with XCode that I didn't have, but I don't remember what it was.  After the curl got eliminated, the project just built, and I didn't need to install anything else, so I'm wondering if the Command Line Tools were a requirement.

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
 06. Symlink the "DRM_Connector_Prerelease" directory to "adobe-rmsdk" within the "Simplified-iOS" directory. (You will need to have obtained the Adobe DRM Connector prerelease from Adobe.)
-07. Follow the instructions in "adobe-rmsdk/RMSDK_User_Manual(obj).pdf" to build OpenSSL (section 12.1) and cURL (section 12.3).*
+07. Build OpenSSL and cURL as described in the following "Building OpenSSL and cURL" section. Ensure you're in the "Simplified-iOS" directory before continuing to the next step.
 08. `sh adobe-rmsdk-build.sh`
```
is it a slash after "rmsdk"?

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
 
-\* If you want to release your app in Apple's App Store, you must add `--enable-ipv6` to line 80 of Adobe's "build.sh" script used for building cURL. Without that flag, cURL will not provide proper support for IPv6 and therefore Apple will reject your application.
+## Building OpenSSL and cURL
+
```
I have a note saved: "don’t gunzip, build.sh script wants to do that for you".

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
+Follow the instructions in "adobe-rmsdk/RMSDK_User_Manual(obj).pdf" to build OpenSSL (section 12.1) and cURL (section 12.3). Be sure to note the following:
+
+* You will need to verify and edit the "build.sh" scripts for both OpenSSL and cURL to reflect the correct version numbers and local directory names (lines 11–24).
+* You must add `--enable-ipv6` to line 80 of Adobe's "build.sh" script used for building cURL. This necessary both due to Apple's requirements for IPv6 support and because the library may not build against recent iOS SDKs otherwise.
```
I have notes saved: "values to edit in build.sh before building:
OPENSSL_VERSION=“1.0.2m”
SDK_VERSION
min_version

tell them to make the public/ios directory to copy the include and lib into"

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
+
+* You will need to verify and edit the "build.sh" scripts for both OpenSSL and cURL to reflect the correct version numbers and local directory names (lines 11–24).
+* You must add `--enable-ipv6` to line 80 of Adobe's "build.sh" script used for building cURL. This necessary both due to Apple's requirements for IPv6 support and because the library may not build against recent iOS SDKs otherwise.
+* cURL 7.57.0 is known _not_ to work and later versions are unlikely to work either. 7.50.0 is recommended.
```
remember to edit the build.sh to match the openassl one

daryanypl commented on README.md:
```diff
@@ -1,18 +1,26 @@
-07. Follow the instructions in "adobe-rmsdk/RMSDK_User_Manual(obj).pdf" to build OpenSSL (section 12.1) and cURL (section 12.3).*
+07. Build OpenSSL and cURL as described in the following "Building OpenSSL and cURL" section. Ensure you're in the "Simplified-iOS" directory before continuing to the next step.
 08. `sh adobe-rmsdk-build.sh`
 09. `cp APIKeys.swift.example Simplified/APIKeys.swift` and edit accordingly.
```
I have a note saved: "for step 9, we need to explain where APIKeys.swift.example is to be coming from", so I'm guessing it didn't just appear in the file, I probably had to download it from a private repository?
